tests/SCF/test_SCF.py

```diff
@@ -1,19 +1,94 @@
 import pytest
 import SCF
+import numpy as np
+from pyscf import gto  # PySCF is a quantum chemistry python module
 
 
-def test_calc_nuclear_repulsion_energy(mol_h2o):
-    assert True
-#    assert SCF.calc_nuclear_repulsion_energy(mol_h2o) == 8.00236706181077,\
-#        "Nuclear Repulsion Energy Test (H2O) Failed"
+def test_all_functions():
+    # Input Data
+    mol_h2o = gto.M(unit="Bohr",
+                    atom="O 0.000000000000  -0.143225816552   0.000000000000;"
+                    + "H 1.638036840407   1.136548822547  -0.000000000000;"
+                    + "H -1.638036840407   1.136548822547  -0.000000000000",
+                    basis='STO-3g')
+    mol_h2o.build()
 
+    E_conv_threshold = 1.0E-10
+    D_conv_threshold = 1.0E-8
+    max_iterations = 1000
 
-def test_calc_initial_density(mol_h2o):
-    """
-    Tests that the initial density returns a zero matrix
-    and tests dimensions
-    """
+    # get the integrals
+    Suv = mol_h2o.intor('int1e_ovlp')  # Overlap Integrals
+    Tuv = mol_h2o.intor('int1e_kin')  # Kinetic Energy 1 electron integrals
+    Vuv = mol_h2o.intor('int1e_nuc')  # Nuclear Repulsion 1 electron integrals
+    eri = mol_h2o.intor("int2e")  # Electron Repulsion 2 electron integrals
+
+    Enuc = SCF.calc_nuclear_repulsion_energy(mol_h2o)
+    assert Enuc == 8.00236706181077, \
+        "Nuclear Repulsion Energy Test (H2O) Failed"
+
+    Huv = SCF.calc_hcore_matrix(Tuv, Vuv)
+    assert np.isclose(Huv[0, 0], -32.57739541261037), "Hcore matrix fail"
+    assert Huv[3, 4] == 0.0, "Hcore matrix fail"
+    assert Huv[4, 3] == 0.0, "Hcore matrix fail"
 
     Duv = SCF.calc_initial_density(mol_h2o)
-    assert Duv.sum() == 0.0
-    assert Duv.shape == (mol_h2o.nao,mol_h2o.nao)
+    for i in range(Duv.shape[0]):
+        for j in range(Duv.shape[1]):
+            assert Duv[i, j] == 0.0, "initial density matrix fail"
+
+    Etot = 0.0
+    for it in range(max_iterations):

```

Comment:
Doing the iterative calculation is not appropriate here, you have created a functional test, not a unit test.